php-classes/Emergence/People/RegistrationRequestHandler.php

```diff
@@ -0,0 +1,178 @@
+<?php
+
+namespace Emergence\People;
+
+use PasswordToken;
+use Emergence\Mailer\Mailer;
+use \ReCaptcha;
+
+class RegistrationRequestHandler extends \RequestHandler
+{
+    // configurables
+    public static $enableRegistration = true;
+    public static $createUser;
+    public static $onRegisterComplete;
+    public static $applyRegistrationData;
+    public static $registrationFields = array(
+        'FirstName'
+        ,'LastName'
+        ,'Gender'
+        ,'BirthDate'
+        ,'Username'
+        ,'Password'
+        ,'Email'
+        ,'Phone'
+        ,'Location'
+        ,'About'
+        
+    );
+    //Initalize Recaptcha
+    
+    // RequestHandler
+    public static $responseMode = 'html';
+    public static $userResponseModes = array(
+        'application/json' => 'json'
+    );
+
+    public static function handleRequest()
+    {
+        switch ($action = static::shiftPath()) {
+            case 'recover':
+                return static::handleRecoverPasswordRequest();
+            case '':
+            case false:
+                return static::handleRegistrationRequest();
+            default:
+                return static::throwNotFoundException();
+        }
+    }
+
+    public static function handleRegistrationRequest($overrideFields = array())
+    {
+       
+       
+        if (!empty($_SESSION['User'])) {
+            return static::throwError('You are already logged in. Please log out if you need to register a new account.');
+        }
+
+        if (!static::$enableRegistration) {
+            return static::throwError('Sorry, self-registration is not currently available. Please contact an administrator.');
+        }
+
+        $filteredRequestFields = array_intersect_key($_REQUEST, array_flip(static::$registrationFields));
+        $additionalErrors = array();
+
+        if (is_callable(static::$createUser)) {
+            $User = call_user_func_array(static::$createUser, array(&$filteredRequestFields, &$additionalErrors));
+        } else {
+            $className = User::getStaticDefaultClass();
+            $User = new $className();
+        }
+
+        if ($_SERVER['REQUEST_METHOD'] == 'POST') {
+            // save person fields
+            $User->setFields(array_merge($filteredRequestFields, $overrideFields));
+             if (empty($_REQUEST['g-recaptcha-response'])) {
```

Comment:
@RowdyChildren you don't need to override this whole class to do this, you can create a file at `php-config/Emergence/People/RegistrationRequestHandler.config.php` that looks like:

```
<?php
Emergence\People\RegistrationRequestHandler::$applyRegistrationData = function($User, $requestData, &$additionalErrors) {
    ...
};
```

and populate `$additionalErrors` from there